daffodil-runtime2/src/main/resources/org/apache/daffodil/runtime2/c/libruntime/infoset.c

```diff
@@ -132,7 +132,8 @@ walkInfosetNode(const VisitEventHandler *handler, const InfosetBase *infoNode)
     const ERD **const childrenERDs = infoNode->erd->childrenERDs;
     const size_t *    offsets = infoNode->erd->offsets;
 
-    for (size_t i = 0; i < count && !error; i++)
+    size_t i;
+    for (i = 0; i < count && !error; i++)
     {
         const size_t offset = offsets[i];
```

Comment:
Is there a setting we can enable so that we can error when we add non-portable code? Even if we only set it for GitHub actions so we can catch it before it's merged if there are concerns about enabling it all time?